packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -583,14 +583,13 @@ class AndroidSdk {
     if (platform.isMacOS) {
       try {
         final String javaHomeOutput = processUtils.runSync(
-          <String>['/usr/libexec/java_home'],
+          <String>['/usr/libexec/java_home', '-v', '1.8'],
           throwOnError: true,
           hideStdout: true,
         ).stdout.trim();
         if (javaHomeOutput != null) {
-          final List<String> javaHomeOutputSplit = javaHomeOutput.split('\n');
-          if ((javaHomeOutputSplit != null) && (javaHomeOutputSplit.isNotEmpty)) {
-            final String javaHome = javaHomeOutputSplit[0].trim();
+          if ((javaHomeOutput != null) && (javaHomeOutput.isNotEmpty)) {
+            final String javaHome = javaHomeOutput.trim();
```

Comment:
it still need to split by lines and javaHome is the last line instead of the first one. This is to account for a case where java 1.8 isn't installed. The first line becomes an error message.

Reply:
Here is how the scenario went here.
I have two versions of java `v1.8` and `v1.13` installed.
Having set `v1.8` as my default version via `jenv`, running `java -version` returns
<img width="471" alt="Screenshot 2020-03-27 at 07 46 39" src="https://user-images.githubusercontent.com/44836329/77729632-76e0a980-6fff-11ea-9b2d-deafc7dec134.png">

Running the original code `/usr/libexec/java_home` returns
<img width="479" alt="Screenshot 2020-03-27 at 07 07 33" src="https://user-images.githubusercontent.com/44836329/77729730-adb6bf80-6fff-11ea-8eff-be4e5e0918c7.png">

The next viable option would have been running `/usr/libexec/java_home -V`, which returns
![Screenshot 2020-03-27 at 07 08 03](https://user-images.githubusercontent.com/44836329/77729824-de96f480-6fff-11ea-838e-d61ef35efc35.png)

Picking the right `java_home` address there is tricky, and as you can see, even the last line has `v1.13`.

Since `flutter` would not accept any other version of `java` except `1.8` (at least for now).
I feel it is okay to try `/usr/libexec/java_home -v 1.8`, which returns
<img width="457" alt="Screenshot 2020-03-27 at 07 11 48" src="https://user-images.githubusercontent.com/44836329/77730001-37ff2380-7000-11ea-8223-3f3e998f7a84.png">
Just what we need and nothing more.